app/models/generic_object_definition.rb

```diff
@@ -1,12 +1,33 @@
 class GenericObjectDefinition < ApplicationRecord
+  TYPE_MAP = {
+    :boolean  => ActiveRecord::Type::Boolean.new,
+    :datetime => ActiveRecord::Type::DateTime.new,
+    :time     => ActiveRecord::Type::Time.new,
+    :float    => ActiveRecord::Type::Float.new,
+    :integer  => ActiveRecord::Type::Integer.new,
+    :string   => ActiveRecord::Type::String.new,
+  }.freeze
+
   validates :name, :presence => true, :uniqueness => true
 
   serialize :properties, Hash
 
   has_one   :picture, :dependent => :destroy, :as => :resource
-  has_many  :generic_objects
+  has_many  :generic_objects, :dependent => :destroy
```

Comment:
If corresponding generic_objects are not deleted they are unless without a definition. Or we can prevent a definition from being deleted if there are still associated generic_objects.